crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs

```diff
@@ -233,15 +233,49 @@ impl OffChainDatabase for OffChainIterableKeyValueView {
         }
     }
 
-    fn balances(
-        &self,
+    fn balances<'a>(
+        &'a self,
         owner: &Address,
-        base_asset_id: &AssetId,
+        start: Option<AssetId>,
+        base_asset_id: &'a AssetId,
         direction: IterDirection,
-    ) -> BoxedIter<'_, StorageResult<(AssetId, TotalBalanceAmount)>> {
+    ) -> BoxedIter<'a, StorageResult<(AssetId, TotalBalanceAmount)>> {
+        match (direction, start) {
+            (IterDirection::Forward, None) => {
+                self.base_asset_first(owner, base_asset_id, direction)
+            }
+            (IterDirection::Forward, Some(asset_id)) => {
+                if asset_id == *base_asset_id {
+                    self.base_asset_first(owner, base_asset_id, direction)
+                } else {
+                    let start = (asset_id != *base_asset_id)
+                        .then_some(CoinBalancesKey::new(owner, &asset_id));
```

Comment:
Why are we redoing the comparaison here as we are already in the `else`
```suggestion
                    let start = Some(CoinBalancesKey::new(owner, &asset_id));
```

Reply:
Nice catch - it's been quite tricky to implement this function correctly, it's probably a leftover from one of the trials.

Fixed in e3e890a5095cb51b5bef0af2316441fea6345006